### PR TITLE
[Merged by Bors] - feat: Set explicit resources on all containers

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of: https://github.com/stackabletech/issues/issues/394

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/42/

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/kafka/pages/usage-guide/storage-resources.adoc [hunk 1]
```diff
@@ -24,21 +24,31 @@ If nothing is configured in the custom resource for a certain role group, then b
 
 include::home:concepts:stackable_resource_requests.adoc[]
 
-If no resource requests are configured explicitly, the Kafka operator uses the following defaults:
+A minimal HA setup consisting of 2 Brokers has the following https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[resource requirements]:
+
+* `1200m` CPU request
+* `3400m` CPU limit
+* `2560Mi` memory request and limit
+* `4Gi` allocated PVs
```
Comment (sbernauer) [inline docs/modules/kafka/pages/usage-guide/storage-resources.adoc:32]:
```suggestion
* `4Gi` persistent storage
```

File: docs/modules/kafka/pages/usage-guide/storage-resources.adoc [hunk 2]
```diff
@@ -24,21 +24,31 @@ If nothing is configured in the custom resource for a certain role group, then b
...
-        resources:
-          memory:
-            limit: '2Gi'
-          cpu:
-            min: '500m'
-            max: '4'
-          storage:
-            log_dirs:
-              capacity: 1Gi
+      replicas: 2
```
Comment (sbernauer) [inline docs/modules/kafka/pages/usage-guide/storage-resources.adoc:53]: I would not mention that here, as we default to a single replica
```suggestion
```

File: tests/templates/kuttl/configuration/20-install-kafka.yaml.j2 [hunk 3]
```diff
@@ -23,36 +23,30 @@ spec:
       resources:
         cpu:
           # Inherit the default value '500m' set by the operator
```
Comment (sbernauer) [inline tests/templates/kuttl/configuration/20-install-kafka.yaml.j2:25]:
```suggestion
          # Inherit the default value '250m' set by the operator
```

File: tests/templates/kuttl/configuration/20-install-kafka.yaml.j2 [hunk 4]
```diff
@@ -23,36 +23,30 @@ spec:
...
-          # min: '500m'
-
+          # min: '250m'
           # Override the default value '4' set by the operator
```
Comment (sbernauer) [inline tests/templates/kuttl/configuration/20-install-kafka.yaml.j2:27]:
```suggestion
          # Override the default value '1' set by the operator
```

File: tests/templates/kuttl/configuration/20-install-kafka.yaml.j2 [hunk 5]
```diff
@@ -23,36 +23,30 @@ spec:
...
-            # Override the default value '1Gi' set by the operator
-            capacity: '3Gi'
+            # Override the default value '2Gi' set by the operator
+            capacity: '1Gi'
     roleGroups:
       default:
         config:
           resources:
             # cpu:
               # Inherit the default value '500m' set by the operator
```
Comment (sbernauer) [inline tests/templates/kuttl/configuration/20-install-kafka.yaml.j2:41]:
```suggestion
              # Inherit the default value '250m' set by the operator
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
Only minor stuff

**Review by sbernauer — APPROVED** [review]
Thanks! LGTM when tests pass

**Comment by maltesander** [comment]
Trying azure:
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/43/

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**